doc/compute-input.md

```diff
@@ -0,0 +1,282 @@
+		{ "name": "hostname", "type": "string"},
+		{ "name": "metric", "type": "string"},
+		{ "name": "status", "type": "string"},
+		{ "name": "monitoring_host", "type": ["null", "string"]},
+		{ "name": "summary", "type": ["null", "string"]},
+		{ "name": "message", "type": ["null", "string"]},
+		{ "name": "tags", "type": [ "null", {
+				"name" : "Tags", "type" : "record", "fields" : [
+					{ "name" : "roc", "type" : ["null", "string"] },
+					{ "name" : "vo", "type" : ["null", "string"] },
+					{ "name" : "vo_fqan", "type" : ["null", "string"] }
+				]
+			}]
+		}
+	]
+}
+```
+
+## Additional input: topology, profiles, factors
+
+This core metric data set is processed and transformed with additional information provided by the argo-connector components. Additional information includes topology, grouping of services, weight factors, lists relevant metrics to be considered, etc. This information is provided per-tenant/per-job in the following path
+
+```
+/var/lib/ar-sync/{tenant-name}/{job-name}
+```
+
+for example for `tenant-name=T1` and `job-name=JobA` the correct path with the sync files is as follows
+
+```
+/var/lib/ar-sync/T1/JobA
+```
+
+Some sync files that concern the whole enviroment such as the downtime information are provided once in the root ar-sync folder `/var/lib/ar-sync/`
+
+<a id="topology"></a>
+
+### Topology files
+
+Topology information is provided by two files:
+
+- groups of endpoints,
+- groups of groups.
+
+A service endpoint is considered by the engine the simplest item of topology. Service endpoint combines the information of `hostname + service_name`. Service endpoints can be grouped together forming upper level entities named endpoint groups. For example an organization's geographical IT site that is being monitored can be considered a group of service endpoints. Information for available endpoint groups is contained in the file group_endpoints.
+
+#### group_endpoints.avro
+
+The file uses avro format and contains the following fields:
+
+| Name | Description | Required |
+|------|-------------|----------|
+|`group`|The name of the group (e.g. MY-SITE-A)|`YES`|
+|`type`| The type of the grouping (e.g. sites)|`YES`|
+|`hostname`| The hostname fqdn part info of the specific endpoint contained in the group|`YES`|
+|`service`| The service name part info of the specific endpoint contained in the group|`YES`|
+|`tags`| user defined tags providing description metadata| `NO`|
+
+Below is the full description of the group_endpoints.avro specification
+
+```json
+{
+	"namespace": "argo.avro",
+	"type": "record",
+	"name": "group_of_service_endpoints",
+		"fields": [
+			{ "name": "type", "type": "string" },
+			{ "name": "group", "type": "string" },
+			{ "name": "service", "type": "string" },
+			{ "name": "hostname", "type": "string" },
+			{ "name": "tags", "type" : ["null", { "name" : "Tags",
+				"type" : "record",
+				"fields" : [
+					{ "name" : "scope", "type" : "string" },
```

Comment:
With commit https://github.com/ARGOeu/argo-egi-connectors/commit/cc89636614ffe9a725ee8b242b9396e2830d4c79 the avro schema field of tags has been changed to 'map' type (actually a generic dictionary):

```json
 {"name": "tags", "type" : ["null", { "name" : "Tags",
                                             "type" : "map",
                                             "values" : ["int", "string"]
                                              }]}
```